### PR TITLE
Add MongoDB, User table, and implement User

### PR DESCRIPTION
**Pull request by adindotdev**
A handful of things to note here:
- We have a secret.js file, ping me for file
- Stopped using the deprecated jsonParser
- Started using `x-www-form-urlencoded`, no changes necessary for the frontend
- It's 500+ lines, let's try to review + merge this ASAP to unblock other ppl

Testing:
- Manually tested all endpoints with Postman

### CODE REVIEW

File: backend/controller/UserController.js [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+ * @param {*} req.params.id the user id
+ * @param {*} req.body.{any} modified value of a userSchema field
+ * @param {*} req.body.addToSet use this to add friendIds and topSongIds
+ * @param {*} req.body.pull use this to remove friendIds and topSongIds
+ */
+exports.updateUserById = (req, res) => {
+  const id = req.params.id;
+  const { addToSet, pull, ...fieldsToUpdate } = req.body;
+  if (pull) fieldsToUpdate.$pull = JSON.parse(pull);
+  if (addToSet) fieldsToUpdate.$addToSet = JSON.parse(addToSet);
```
Comment (ianmah) [inline backend/controller/UserController.js:68]: what is the `$` syntax

File: backend/schema/User.js [hunk 2]
```diff
@@ -0,0 +1,16 @@
+const mongoose = require('mongoose');
+
+const userSchema = new mongoose.Schema(
+  {
+    name: {
```
Comment (ianmah) [inline backend/schema/User.js:5]: name or id?

**Review by ianmah — APPROVED** [review]
is this how fangineers write code godanm this 🅱️razy!

File: backend/controller/UserController.js (hunk 1 above)
Comment (adindotdev) [reply backend/controller/UserController.js:68]: It's an update operator that denotes a set of actions. Without `$`, MongoDB will think it's a `key: value` pair to be updated.

File: backend/schema/User.js (hunk 2 above)
Comment (adindotdev) [reply backend/schema/User.js:5]: ID is autogenerated as `_id`. Name is the user's actual name.

**Review by vandyliu — APPROVED** [review]
sheeesh

File: backend/schema/User.js (hunk 2 above)
Comment (ianmah) [reply backend/schema/User.js:5]: i mean like spotify id or email or something

**Pull request merged**